src/components/super-cursor.js

```diff
@@ -48,10 +62,20 @@ AFRAME.registerComponent("super-cursor", {
     this.isGrabbing = this.data.cursor.components["super-hands"].state.has("grab-start");
     let isIntersecting = false;
 
+    const camera = this.data.camera.components.camera.camera;
+    const raycaster = this.el.components.raycaster.raycaster;
+    raycaster.setFromCamera(this.mousePos, camera);
+    this.origin = raycaster.ray.origin;
+    this.direction = raycaster.ray.direction;
+    this.el.setAttribute("raycaster", { origin: this.origin, direction: this.direction });
+
+    let className = null;
+
     if (!this.isGrabbing) {
       const intersections = this.el.components.raycaster.intersections;
       if (intersections.length > 0 && intersections[0].distance <= this.data.maxDistance) {
         isIntersecting = true;
```

Comment:
Instead of storing `isIntersecting`, `className`, `point` and `currentDistance` I would probably just store the interesection itself.